File: P1.3/P1.3_opcional/FilterManager.dart
import 'Filter.dart';
import 'FilterChain.dart';
import 'Target.dart';

class FilterManager {
  late FilterChain _filterChain;
  Target _target;

  FilterManager(this._target) {
    _filterChain = FilterChain();
  }

  addFilter(Filter f) => _filterChain.addFilter(f);

  filterRequest() {
      _filterChain.execute(_target);
      _target.execute();
  }
}

File: P1.3/P1.3_obligatorio/src/FilterManager.cpp
#include <iostream>
#include "FilterManager.h"

FilterManager::FilterManager(Target*  t) {
    this->target = std::unique_ptr<Target>(t);
}

void FilterManager::addFilter(Filter* & f) {
    this->filterChain.addFilter(f);
}

void FilterManager::filterRequest() {
    this->filterChain.execute(*target);
    this->target->execute();
}